luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -121,7 +124,7 @@ end
 local function preventOverkillDamage(unitID, damage, health, healthRatioMultiplier)
 	damage = damage * healthRatioMultiplier
 	if damage >= health then
-		spDestroyUnit(unitID) --this ensures a wreck is left behind. If damage is too great, it destroys the heap.
+		fallingKillQueue[unitID] = true --done in GameFrame to take it out of unitPreDamaged
```

Comment:
Instead of manually maintaining a queue you could just do `Script.DelayByFrames(1, spDestroyUnit, unitID)`

Reply:
Where are these script functions defined? Does it help with the bottleneck that occurs from calling a function over and over again from Lua scripts?